go/pkg/sysdb/coordinator/coordinator.go

```diff
@@ -83,8 +95,19 @@ func (s *Coordinator) GetCollections(ctx context.Context, collectionID types.Uni
 	return s.catalog.GetCollections(ctx, collectionID, collectionName, tenantID, databaseName, limit, offset)
 }
 
+func (s *Coordinator) GetSoftDeletedCollections(ctx context.Context, collectionID *string, tenantID string, databaseName string, limit int32) ([]*model.Collection, error) {
+	return s.catalog.GetSoftDeletedCollections(ctx, collectionID, tenantID, databaseName, limit)
+}
+
 func (s *Coordinator) DeleteCollection(ctx context.Context, deleteCollection *model.DeleteCollection) error {
-	return s.catalog.DeleteCollection(ctx, deleteCollection)
+	if s.deleteMode == SoftDelete {
+		return s.catalog.DeleteCollection(ctx, deleteCollection, true)
+	}
+	return s.catalog.DeleteCollection(ctx, deleteCollection, false)
+}
+
+func (s *Coordinator) CleanupSoftDeletedCollection(ctx context.Context, deleteCollection *model.DeleteCollection) error {
```

Comment:
Why introduce a separate RPC?  Wouldn't calling with deleteMode set to hard delete be sufficient?